gateways/token_balances_api_gateway.py

```diff
@@ -0,0 +1,111 @@
+from typing import List, Optional
+
+from elasticsearch import Elasticsearch
+
+from common.configuration import ELASTIC_RESULTS_PER_PAGE, ELASTIC_SEARCH_TIMEOUT, ELASTIC_TOKEN_BALANCES_INDEX
+from gateways.clients.elastic_search_client import ElasticSearchClient
+from utils.elastic_search.elastic_search_utils import ElasticSearchUtils
+
+SORTABLE_FIELDS = {
+    'address': 'keyword',
+    'token_id': 'keyword',
+    'unlocked_balance': 'long',
+    'locked_balance': 'long',
+    'total': 'long',
+}
+
+
+class TokenBalancesApiGateway:
+    """ Gateway to interact with the Hathor ElasticSearch cluster with the token balances index
+    """
+
+    def __init__(
+        self,
+        elastic_search_client: Optional[Elasticsearch] = None,
+    ) -> None:
+        self.elastic_search_client = ElasticSearchClient(elastic_index=ELASTIC_TOKEN_BALANCES_INDEX,
```

Comment:
Need to check if `elastic_search_client` was already passed as parameter of the function

Reply:
This check is already done in `ElasticSearchClient`, `client` is optional, it will be created if it is not passed

This is the same behavior done in the token list gateway